core/src/main/java/com/datastax/oss/driver/internal/core/os/NativeImpl.java

```diff
@@ -15,13 +15,15 @@
  */
 package com.datastax.oss.driver.internal.core.os;
 
-public interface NativeImpl {
+import java.util.Optional;
 
-  public boolean gettimeofdayAvailable();
+public interface NativeImpl {
 
-  public long gettimeofday();
+  /* Maintained to allow Native.isXAvailable() functionality without trying to make a native call if
+   * the underlying support _is_ available. */
+  public boolean available();
 
-  public boolean getpidAvailable();
+  public Optional<Long> gettimeofday();
```

Comment:
Why not `OptionalLong` and `OptionalInt`?

Reply:
Because OptionalLong and OptionalInt do not subclass Optional; they are their own type hierarchy.  As a result the map/flatMap ops off Optional<POSIX> can't generate these types.

Reply:
Oh you are flatmapping them down the line. Ok then.